src/clp_ffi_py/ir/native/PyKeyValuePairLogEvent.cpp

```diff
@@ -0,0 +1,697 @@
+                       py_reinterpret_cast<PyObject>(m_parent),
+                       m_schema_tree_node->get_key_name().data(),
+                       py_reinterpret_cast<PyObject>(m_py_dict.get())
+               );
+    }
+
+    /**
+     * Releases the underlying Python dictionary as the root dictionary to return.
+     * @return The released Python dictionary on success.
+     * @return nullptr on failure with the relevant Python exception and error set.
+     */
+    [[nodiscard]] auto release_root() -> PyDictObject* {
+        if (false == is_root()) {
+            PyErr_SetString(
+                    PyExc_RuntimeError,
+                    "KeyValuePairLogEvent.to_dict(): only root can be released"
+            );
+            return nullptr;
+        }
+        return m_py_dict.release();
+    }
+
+    [[nodiscard]] auto get_py_dict() -> PyDictObject* { return m_py_dict.get(); }
+
+    [[nodiscard]] auto is_root() const -> bool { return m_schema_tree_node->is_root(); }
+
+private:
+    // Constructor
+    PyDictSerializationIterator(
+            SchemaTree::Node const* schema_tree_node,
+            std::vector<SchemaTree::Node::id_t> child_schema_tree_nodes,
+            PyDictObject* parent,
+            PyObjectPtr<PyDictObject> py_dict
+    )
+            : m_schema_tree_node{schema_tree_node},
+              m_child_schema_tree_nodes{std::move(child_schema_tree_nodes)},
+              m_child_schema_tree_node_it{m_child_schema_tree_nodes.cbegin()},
+              m_parent{parent},
+              m_py_dict{std::move(py_dict)} {}
+
+    SchemaTree::Node const* m_schema_tree_node;
+    std::vector<SchemaTree::Node::id_t> m_child_schema_tree_nodes;
+    std::vector<SchemaTree::Node::id_t>::const_iterator m_child_schema_tree_node_it;
+    PyDictObject* m_parent;
+    PyObjectPtr<PyDictObject> m_py_dict;
+};
+
+/**
+ * Callback of `PyKeyValuePairLogEvent`'s `__init__` method.
+ */
+// NOLINTNEXTLINE(cppcoreguidelines-avoid-c-arrays)
+PyDoc_STRVAR(
+        cPyKeyValuePairLogEventDoc,
+        "This class represents a key-value pair log event and provides ways to access the"
+        " underlying log data. This class is designed to be instantiated by the IR deserializer."
+        " However, direct instantiation using the `__init__` method is also supported for testing"
+        " purposes, although this may not be as efficient as emitting from the IR deserializer.\n\n"
+        "The signature of `__init__` method is shown as following:\n\n"
+        "__init__(self, dictionary)\n\n"
+        "Initializes an object that represents a key-value pair log event from the given Python"
+        " dictionary. Notice that each object should be strictly initialized only once. Double"
+        " initialization will result in memory leak.\n\n"
+        ":param dictionary: A dictionary representing the key-value log event, where all keys are"
+        " expected to be of string type, including keys inside any sub-dictionaries.\n"
```

Comment:
```suggestion
        "This class represents a key-value pair log event and provides methods to access the"
        " key-value pairs. This class is designed to be instantiated by the IR deserializer."
        " However, direct instantiation using the `__init__` method is also supported for testing"
        " purposes, although this may not be as efficient as emission from the IR deserializer.\n\n"
        "__init__(self, dictionary)\n\n"
        "Initializes a :class:`KeyValuePairLogEvent` from the given Python dictionary. Note that"
        " each object should only be initialized once. Double initialization will result in a"
        " memory leak.\n\n"
        ":param dictionary: A dictionary representing the key-value pair log event, where all keys"
        " must be strings, including keys inside any sub-dictionaries.\n"
```